libraries-bom-protobuf3/pom.xml

```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0                              http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <!-- adding comment for testing-->
+  <groupId>com.google.cloud</groupId>
+  <artifactId>libraries-bom-protobuf3</artifactId>
+  <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:libraries-bom-protobuf3:current} -->
+  <packaging>pom</packaging>
+
+  <parent>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>libraries-bom</artifactId>
+    <version>26.48.0-SNAPSHOT</version><!-- {x-version-update:libraries-bom:current} -->
+    <relativePath>../libraries-bom</relativePath>
+  </parent>
+
+  <name>Google Cloud Platform Supported Libraries Protobuf3</name>
+  <description>
+    A compatible set of Google Cloud open source libraries. This bom is compatible with protobuf-java 3.x versions.
+    Document: https://cloud.google.com/java/docs/bom
+  </description>
+  <url>https://cloud.google.com/java/docs/bom</url>
+  <organization>
+    <name>Google LLC</name>
+    <url>https://cloud.google.com</url>
+  </organization>
+  <inceptionYear>2024</inceptionYear>
+  <developers>
+    <developer>
+      <name>Cloud Java team</name>
+    </developer>
+  </developers>
+
+  <distributionManagement>
+    <snapshotRepository>
+      <id>sonatype-nexus-snapshots</id>
+      <url>https://google.oss.sonatype.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+    <repository>
+      <id>sonatype-nexus-staging</id>
+      <url>https://google.oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+    </repository>
+  </distributionManagement>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.protobuf</groupId>
+        <artifactId>protobuf-bom</artifactId>
+        <version>3.25.5</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <profiles>
+    <profile>
```

Comment:
Profiles can not extended from parent poms, hence have to be duplicated.

Reply:
Really? Java-shared-config (the parent of many of our projects) has profiles for the projects.

Can you write down your observation?

Reply:
The difference here is that the parent is a BOM with scope import, which I believe does not include profiles. 
To avoid duplications, is it possible to create a separate parent POM that defines these common profiles? This libraries-bom-protobuf3  can then inherit from both the libraries-bom and the parent POM (for profiles and other settings).

Reply:
@suztomo I learned from the [official Maven doc](https://maven.apache.org/pom.html#inheritance) and [other's experience](https://www.dashbay.com/2011/03/maven-profile-inheritance/) that profiles are not inherited. How did we know the profiles are inherited in other cases? Plugins are inherited, so maybe it was using the default plugins not the plugin overridden in the profiles?

@zhumin8 libraries-bom-protobuf3 is inherited from libraries-bom, not importing libraries-bom. Creating a separate base bom would have the same problem if profiles can not be inherited. In addition, we want to introduce no change to libraries-bom and no additional layers if possible, as we already have a lot layers in the release process.

Reply:
https://maven.apache.org/pom.html#inheritance has this line for elements not inherited:

> profiles (but the effects of active profiles from parent POMs are)

It means the profiles are not inherited but the effects of active profiles from parent POMs are inherited. Thus the effect of the parent pom (the plugin configurations in our case) is inherited.

Reply:
So in this case when we call `mvn deploy -Prelease` or activate the profile by `performRelease` property under the root folder of `java-cloud-bom`, each submodule would use the plugins overridden in the parent pom?

Reply:
Yes, I read so. You can validate it with `mvn help:effective-pom -Dverbose`.